android/expoview/src/main/java/host/exp/exponent/fcm/ExpoFirebaseMessagingDelegate.kt

```diff
@@ -0,0 +1,89 @@
+package host.exp.exponent.fcm
+
+import android.content.Context
+import com.google.firebase.messaging.RemoteMessage
+import expo.modules.notifications.notifications.model.NotificationContent
+import expo.modules.notifications.notifications.model.NotificationRequest
+import expo.modules.notifications.notifications.model.triggers.FirebaseNotificationTrigger
+import expo.modules.notifications.service.delegates.FirebaseMessagingDelegate
+import host.exp.exponent.ABIVersion
+import host.exp.exponent.Constants
+import host.exp.exponent.analytics.EXL
+import host.exp.exponent.kernel.ExperienceId
+import host.exp.exponent.notifications.PushNotificationHelper
+import host.exp.exponent.notifications.model.ScopedNotificationRequest
+import host.exp.exponent.storage.ExponentDB
+import org.json.JSONException
+import org.json.JSONObject
+
+class ExpoFirebaseMessagingDelegate(context: Context) : FirebaseMessagingDelegate(context) {
+  override fun onNewToken(token: String) {
+    if (!Constants.FCM_ENABLED) {
+      return
+    }
+    super.onNewToken(token)
+    FcmRegistrationIntentService.registerForeground(context.applicationContext, token)
+  }
+
+  override fun onMessageReceived(remoteMessage: RemoteMessage) {
+    if (!Constants.FCM_ENABLED) {
+      return
+    }
+    val experienceDBObject = ExponentDB.experienceIdToExperienceSync(remoteMessage.data["experienceId"])
+    if (experienceDBObject != null) {
```

Comment:
```suggestion
    experienceDBObject?.let {
```
more kotlin-styled 😉